### PR TITLE
feat(ia): Settings / Connections - Plugins & Analytics

### PR DESCRIPTION
**Pull request by jaredrethman**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

* Migrated from _/wp-admin/admin.php?page=newspack-connections-wizard_.
* Adds Plugin and Analytics sections/components.
* Refactored API Fetch to use new `useWizardApiFetch` hook, introduced in #3163.
* Refactored `ActionCard` components to use `WizardActionCard` & `WizardPluginCard`.
* Added type definitions to `useObjectState` to appease TypeScript.
* Includes major refactor introduced in #3188 

### How to test the changes in this Pull Request:

**Original locations:**
* Plugins: _/wp-admin/admin.php?page=newspack-connections-wizard_
* Analytics: _/wp-admin/admin.php?page=newspack-analytics-wizard_

1. Checkout `feat/ia-settings-connections__plugins` and build assets.
2. Navigate to _/wp-admin/admin.php?page=newspack-settings_

**Test 1:**
1. Wait for relevant sections to load.
2. Observe plugins and analytics load as expected. Can be compared to legacy instance of Plugins and Analytics sections - _/wp-admin/admin.php?page=newspack-connections-wizard_.

**Test 2:**
1. Run `npm run watch`.
2. Navigate to `assets/wizards/newspack/views/settings/connections/plugins.tsx`.
3. Enforce an error: Update `path` `'/newspack/v1/plugins/jetpack'` with `'/newspack/v1/plugins/jetpacks'`. Update `slug` `jetpack` with `jetpacks`.
4. Refresh browser and confirm error is present.
Screenshot 2024-06-28 at 11 21 06

**Test 3: (X represents the plugins name)**
1. Pick a plugin and make sure its uninstalled.
2. When observing this plugin within the context of this PR. Observe that there is a "Install X" call to action present.
3. Click `Install X`, wait for completion, then, confirm the plugin installs correctly and the CTA has changed to "Activate X"
4. Click "Activate X", wait for completion, then confirm the page reloads.
5. Now observe that the plugin either has a green check (signaling all is okay and no further action is required) or a "Complete Setup" CTA is present.
6. If "Complete Setup" CTA is shown click it and confirm it takes you to the relevant settings page.

**Additional steps:**
- Confirm data (including errors) are persistent between tab changes.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/newspack/views/settings/connections/plugins.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ */
+import WizardsPluginCard from '../../../../wizards-plugin-card';
+
+const { plugins: pluginsSection, analytics: analyticsSection } =
+	window.newspackSettings.connections.sections;
+
+const PLUGINS: Record< string, PluginCard > = {
+	jetpack: {
+		slug: 'jetpack',
+		path: '/newspack/v1/plugins/jetpack',
```
Comment (chickenn00dle) [inline assets/wizards/newspack/views/settings/connections/plugins.tsx]: Looks like these paths aren't actually being used by `WizardsPluginCard`. The fetch path seems to be determined by `slug` here: https://github.com/Automattic/newspack-plugin/blob/22f9be9c3bd466bd8274c304b09c7bef03a0eb5e/assets/wizards/wizards-plugin-card.tsx#L27-L38

According to this testing step, it seems as though we are expecting this to be used though:

```
Enforce an error: Update path '/newspack/v1/plugins/jetpack' with '/newspack/v1/plugins/jetpacks'. Update slug jetpack with jetpacks.
```

**Review by chickenn00dle — CHANGES_REQUESTED** [review]
Hey @jaredrethman,

I tested and was able to confirm the connected state, as well as the error state, but had trouble with the uninstall state.

I have everlit uninstalled:
Screenshot 2024-06-28 at 15 00 33

But it appears to be connected:
Screenshot 2024-06-28 at 15 00 42

**Comment by jaredrethman** [comment]
> Hey @jaredrethman,
> 
> I tested and was able to confirm the connected state, as well as the error state, but had trouble with the uninstall state.

@chickenn00dle thank for the feedback, this is a odd one.

I'm unable to reproduce this. I've tested the following two scenarios:
1. Completing install + activation, then deactivating, then deleting.
2. Completing install + activation, then deleting without deactivating first. 

And in both cases the outcome was the same:
Screenshot 2024-07-05 at 21 58 21

Could you please share some minimal reproducible steps and/or test the latest changes and see if the latest code fixes it.

File: assets/wizards/newspack/views/settings/connections/plugins.tsx (hunk 1 above)
Comment (jaredrethman) [reply assets/wizards/newspack/views/settings/connections/plugins.tsx]: Nice catch, `path` became redundant when I introduced Everlit, I missed the housekeeping step. Have removed `path` in 71ccbd5c5cc8122adcd265b39f514075023de1a1

**Comment by chickenn00dle** [comment]
> Could you please share some minimal reproducible steps and/or test the latest changes and see if the latest code fixes it.

I pulled the latest changes, but am still seeing the problem 😞 

This seems to happen when the plugin begins as active and connected, but then is disconnected. So to reproduce for Site Kit for example:

1. Make sure site kit is active and connected
2. Pull changes and check connection settings via new settings page
3. Go to plugins and deactivate Site Kit
4. Go back to the connections settings and refresh
5. Site kit appears to still be active

Here is a screen recording if that helps:


https://github.com/Automattic/newspack-plugin/assets/17905991/7da5190d-b8b4-42e3-b480-b806bfef402f

**Comment by jaredrethman** [comment]
> > Could you please share some minimal reproducible steps and/or test the latest changes and see if the latest code fixes it.
> 
> I pulled the latest changes, but am still seeing the problem 😞
> 
> This seems to happen when the plugin begins as active and connected, but then is disconnected. So to reproduce for Site Kit for example:
> 
> 1. Make sure site kit is active and connected
> 2. Pull changes and check connection settings via new settings page
> 3. Go to plugins and deactivate Site Kit
> 4. Go back to the connections settings and refresh
> 5. Site kit appears to still be active
> 
> Here is a screen recording if that helps:
> 
>  Screen.Recording.2024-07-08.at.11.59.59.mov

Thanks for your response @chickenn00dle , can you confirm the same behavior when `WP_NEWSPACK_DEBUG` const is set to `false`?

If the problem disappears it's because of this code:
https://github.com/Automattic/newspack-plugin/blob/589a4f9f99e761c07339da9aecc9664b74452cae/includes/class-plugin-manager.php#L390-L392

If the problem still persists, we can look into next steps.

**Review by chickenn00dle — APPROVED** [review]
> If the problem still persists, we can look into next steps.

Aha! Good catch. This was the problem!

Screenshot 2024-07-08 at 17 27 41


Looks good now. Giving this a thumbs up assuming conflicts will be resolved first 👍

**Pull request merged**